books/RayTracingTheNextWeek.html

```diff
@@ -2256,7 +2256,319 @@
+on the plane. We've already been using a coordinate frame for our 3D space -- this is defined by an
+origin point $\mathbf{O}$ and three basis vectors $\mathbf{x}$, $\mathbf{y}$, and $\mathbf{z}$.
+
+Since a plane is a 2D construct, we just need a plane origin point $\mathbf{O}$ and _two_ basis
+vectors: $\mathbf{u}$ and $\mathbf{v}$. Now normally, you see axes that are perpendicular to each
+other, so they all meet at 90° angles. However, this doesn't need to be the case in order to span
+the entire space -- you just can't have two axes that are parallel to each other.
+
+  ![Figure [ray-plane]: Ray-plane intersection](../images/fig-2.06-ray-plane.jpg)
+
+Consider figure [ray-plane] as an example. Ray $\mathbf{R}$ intersects the plane, yielding
+intersection point $\mathbf{P}$. Measuring against plane vectors $\mathbf{u}$ and $\mathbf{v}$, the
+intersection point $\mathbf{P}$ is at $\mathbf{O} + (1)\mathbf{u} + (\frac{1}{2})\mathbf{v}$. In
+other words, the plane coordinates of intersection point $\mathbf{P}$ is $(1,\frac{1}{2})$.
+
+Generally, given some arbitrary point $\mathbf{P}$, we seek two scalar values $\alpha$ and $\beta$,
+so that
+
+  $$ \mathbf{P} = \mathbf{O} + \alpha \mathbf{u} + \beta \mathbf{v} $$
+
+If $\mathbf{u}$ and $\mathbf{v}$ were guaranteed to be orthogonal to each other (forming a 90°
+angle between them), then this would be a simple matter of projecting $\mathbf{P}$ onto the two
+basis vectors $\mathbf{u}$ and $\mathbf{v}$. However, since they can form any angle (other than
+zero), the math's a little bit tricker.
+
+$$ \mathbf{P} = \mathbf{O} + \alpha \mathbf{u} + \beta \mathbf{v}$$
+
+$$ \mathbf{p} = \mathbf{P} - \mathbf{O} = \alpha \mathbf{u} + \beta \mathbf{v} $$
+
+Here, $\mathbf{P}$ is the _point_ of intersection, and $\mathbf{p}$ is the _vector_ from
+$\mathbf{O}$ to $\mathbf{P}$.
+
+Cross the above equation with $\mathbf{u}$ and $\mathbf{v}$, respectively:
+
+$$ \begin{align*}
+   \mathbf{u} \times \mathbf{p} &= \mathbf{u} \times (\alpha \mathbf{u} + \beta \mathbf{v}) \\
+   &= \mathbf{u} \times \alpha \mathbf{u} + \mathbf{u} \times \beta \mathbf{v} \\
+   &= \alpha(\mathbf{u} \times \mathbf{u}) + \beta(\mathbf{u} \times \mathbf{v})
+   \end{align*} $$
+
+$$ \begin{align*}
+   \mathbf{v} \times \mathbf{p} &= \mathbf{v} \times (\alpha \mathbf{u} + \beta \mathbf{v}) \\
+   &= \mathbf{v} \times \alpha \mathbf{u} + \mathbf{v} \times \beta \mathbf{v} \\
+   &= \alpha(\mathbf{v} \times \mathbf{u}) + \beta(\mathbf{v} \times \mathbf{v})
+   \end{align*} $$
+
+Since any vector crossed with itself yields zero, these equations simplify to
+
+$$ \mathbf{u} \times \mathbf{p} = \beta(\mathbf{u} \times \mathbf{v}) $$
+$$ \mathbf{v} \times \mathbf{p} = \alpha(\mathbf{v} \times \mathbf{u}) $$
+
+To solve for the coefficients $\alpha$ and $\beta$, we can take the dot product of both sides of the
+above equations with the plane normal $\mathbf{n} = \mathbf{u} \times \mathbf{v}$, reducing both
```

Comment:
Are we restricting u and v to unit length? Because we normalized the normal vector above, so this equation isn't strictly true unless the cross and u and v just happens to be unit length.

Reply:
No, *u* and *v* are not restricted to unit length, and the normal vector *n* is merely *u* ⨯ *v*. No normalization needed; we're just cancelling terms.

Reply:
Of course, correct me if this is not wrong, but I do remember verifying on paper with non-unit vectors.